From 8116f8d7a15ffa03b08cfd9c7129a15ff2b7b6ec Mon Sep 17 00:00:00 2001
From: Adrian Alonso <aalonso@freescale.com>
Date: Tue, 20 Nov 2012 18:25:40 -0600
Subject: [PATCH 2277/2463] ENGR00234354-2: board-mx6q_sabreauto aline
 weim-nor partition layout

* Aline weim-nor partition layout with u-boot expected
  offtsets
  "bootloader"  /dev/mtd0
  "bootenv"     /dev/mtd1
  "kernel"      /dev/mtd2
  "rootfs"      /dev/mtd3

Signed-off-by: Adrian Alonso <aalonso@freescale.com>
---
 arch/arm/mach-mx6/board-mx6q_sabreauto.c |   18 +++++++++++++++---
 1 files changed, 15 insertions(+), 3 deletions(-)

diff --git a/arch/arm/mach-mx6/board-mx6q_sabreauto.c b/arch/arm/mach-mx6/board-mx6q_sabreauto.c
index ee8935f..95b9e0a 100644
--- a/arch/arm/mach-mx6/board-mx6q_sabreauto.c
+++ b/arch/arm/mach-mx6/board-mx6q_sabreauto.c
@@ -444,17 +444,29 @@ static void spi_device_init(void)
 	spi_register_board_info(m25p32_spi0_board_info,
 				ARRAY_SIZE(m25p32_spi0_board_info));
 }
+
 static struct mtd_partition mxc_nor_partitions[] = {
 	{
-		.name	= "Bootloader",
+		.name	= "bootloader",
 		.offset	= 0,
-		.size	=  0x00080000,
+		.size	= SZ_256K,
+		.mask_flags = MTD_WRITEABLE,
 	}, {
-		.name	= "nor.Kernel",
+		.name = "bootenv",
+		.offset = MTDPART_OFS_APPEND,
+		.size = SZ_256K,
+		.mask_flags = MTD_WRITEABLE,
+	}, {
+		.name	= "kernel",
 		.offset	= MTDPART_OFS_APPEND,
+		.size	= SZ_4M,
+	}, {
+		.name	= "rootfs",
+		.offset = MTDPART_OFS_APPEND,
 		.size	= MTDPART_SIZ_FULL,
 	},
 };
+
 static struct resource nor_flash_resource = {
 	.start		= CS0_BASE_ADDR,
 	.end		= CS0_BASE_ADDR  +  0x02000000 - 1,
-- 
1.7.7.4

